pkg/dockerfilegen/generator.go

```diff
@@ -152,9 +152,27 @@ func generateDockerfile(params Params, mainPackagesPaths sets.Set[string]) error
 		goVersion = strings.Join(strings.Split(goVersion, ".")[0:2], ".")
 	}
 
+	metadata, err := project.ReadMetadataFile(params.ProjectFilePath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("%w: Failed to read project metadata file: %w",
+				ErrBadConf, errors.WithStack(err))
+		}
+		log.Println("File not found:", params.ProjectFilePath, "(Using defaults)")
+		metadata = project.DefaultMetadata()
+	}
+
+	rhelVersion := "rhel-8"
+	minorVersion, err := strconv.Atoi(strings.Replace(metadata.Project.Tag, "knative-v1.", "", 1))
```

Comment:
Is metadata.Project.Tag coming from (SO) project.yaml? I don't see it project.yaml